test/e2e/util.go

```diff
@@ -2098,7 +2098,7 @@ func NewHostExecPodSpec(ns, name string) *api.Pod {
 	pod := &api.Pod{
 		TypeMeta: unversioned.TypeMeta{
 			Kind:       "Pod",
-			APIVersion: latest.GroupOrDie("").GroupVersion.Version,
```

Comment:
Right. I'm saying don't put the constant in the install package, and don't
move the install guts into a package that's not called install. It should
be very obvious and controlable to the person putting together a binary
which groups (and in the future, versions) that binary will have the
ability to understand. Among other things, we'll need this to generate
clients (and servers) for testing which do or don't understand certain
groups/versions.

On Tue, Dec 8, 2015 at 4:35 AM, David Eads <notifications@github.com> wrote:

> In test/e2e/util.go
> <https://github.com/kubernetes/kubernetes/pull/18154#discussion_r46947421>
> :
>
> > @@ -2098,7 +2098,7 @@ func NewHostExecPodSpec(ns, name string) *api.Pod {
> >  	pod := &api.Pod{
> >  		TypeMeta: unversioned.TypeMeta{
> >  			Kind:       "Pod",
> > -			APIVersion: latest.GroupOrDie("").GroupVersion.Version,
>
> To be extra clear, if mygrouppackage installs the codec & schema etc when
> imported, I would not be fine with that.
>
> pkg/apis/<group>/install does exactly that today. See here:
> https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/componentconfig/install/install.go#L53
> and here:
> https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/componentconfig/install/install.go#L73
> (scheme registered down in that method). How were you thinking it was done?
> install is different/special?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18154/files#r46947421>.
>